File: apps/web/src/api/vocabulary.ts
import request from '@/utils/request';

export interface VocabularyWord {
  id: number;
  word: string;
  uk_phonetic: string | null;
  us_phonetic: string | null;
  meaning: string;
  article_id: number | null;
  article_title: string | null;
  created_at: string;
}

export interface TimelineGroup {
  date: string;
  words: VocabularyWord[];
}

export interface VocabularyTimelineResponse {
  timeline: TimelineGroup[];
  total: number;
}

/**
 * 获取生词本时间线
 */
export async function getVocabularyTimeline(): Promise<VocabularyTimelineResponse> {
  return request.get('/vocabularies/timeline');
}

File: apps/backend/app/schemas/vocabulary.py
"""
生词本响应模型。
"""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field


class VocabularyResponse(BaseModel):
    id: int
    word: str
    uk_phonetic: str | None
    us_phonetic: str | None
    meaning: str
    article_id: int | None
    article_title: str | None = None
    created_at: datetime

    model_config = {"from_attributes": True}


class TimelineGroup(BaseModel):
    date: str = Field(..., description="日期，格式 YYYY-MM-DD")
    words: list[VocabularyResponse]


class VocabularyTimelineResponse(BaseModel):
    timeline: list[TimelineGroup]
    total: int
